### DISCUSSION TITLE
Is it good to add an arg "use_thread" in `asyncify` function?

### DISCUSSION THREAD

**Discussion by monchin** [fastapi/asyncer#168] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Asyncer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Asyncer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Asyncer but to [AnyIO](https://github.com/agronholm/anyio).
- [X] I already checked if it is not related to Asyncer but to [Trio](https://github.com/python-trio/trio).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

Hi here.
I have read [asyncer's doc about `asyncify`](https://asyncer.tiangolo.com/tutorial/asyncify/), it's really convenient. But the doc just said 

> asyncify() takes the sync (blocking) function that you want to call and then returns another async function that takes the actual arguments for the original sync function.
> Then you can await that and get the return value that was safely executed in a "worker thread" without blocking the event loop.

If I didn't see the second paragraph above, I would believe that any sync function could be asyncified by `asyncify`. So I read the code of `asyncify`

```py
    async def wrapper(
        *args: T_ParamSpec.args, **kwargs: T_ParamSpec.kwargs
    ) -> T_Retval:
        partial_f = functools.partial(function, *args, **kwargs)
        return await anyio.to_thread.run_sync(
            partial_f, cancellable=cancellable, limiter=limiter
        )
```
It uses `anyio.to_thread.run_sync`

### Description

If the sync function is IO-intensive, `asyncify` would be great. But if the sync function is CPU-intensive, `asyncify` would not work because of GIL. For this situation, `anyio` offers [`to_process`](https://anyio.readthedocs.io/en/stable/api.html#anyio.to_process.run_sync). 
So my point is, if it is good to offer a bool arg `use_thread` in `asyncify` to make sure whether use `to_thread` or `to_process`? Or offer a new function `asyncify_to_process`? At lease I think now the doc should show that `asyncify` would be appropriate only when the sync function is IO-intensive.

### Operating System

Linux

### asyncer Version

0.0.7

### Python Version

3.10.12

**Comment by YuriiMotov** [comment, marked as answer]
Yes, currently docs don't cover the topic of performance optimization.

But the following statement is not correct

> But if the sync function is CPU-intensive, asyncify would not work because of GIL.

It will work - your sync function is CPU-intensive will not block the event loop and this is what `asyncify` is intended to solve.

It might be a good idea to add a section in docs describing use cases that are better to solve with other approaches

**Reply by monchin** [reply]
Thank you, I misundeestood the mechanism of `asyncify`.
